### PR TITLE
1300 Make collection element insertion idempotent

### LINKED ISSUE #1300: Collection insertions are non-idempotent after migration
**What Needs to be Done?**

bvh does multiple insertions of the same index expecting them to be idempotent. This works out if the load balancer doesn't run because it does a local check to ensure not to insert multiple times. However, with the load balancer, two copies of the element end up existing causing a crash.

The solution is to query the home node to ensure that the insertion has not happened already, even if the insertion is off home (the user is allowed to insert wherever they desire).

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #1300

### CODE REVIEW

File: src/vt/topos/location/location.impl.h [hunk 1]
```diff
@@ -180,6 +180,35 @@ void EntityLocationCoord<EntityID>::registerEntity(
   }
 }
 
+template <typename EntityID>
+void EntityLocationCoord<EntityID>::registerEntityRemote(
+  EntityID const& id, NodeType const& home, NodeType const create_node,
+  LocMsgActionType msg_action
+) {
+  auto reg_iter = local_registered_.find(id);
```
Comment (PhilMiller) [inline src/vt/topos/location/location.impl.h:188]: Should we be asserting that `home == this_node`?

**Review by PhilMiller — APPROVED** [review]
Looks good

File: src/vt/topos/location/location.impl.h (hunk 1 above)
Comment (lifflander) [reply src/vt/topos/location/location.impl.h:188]: Yes, good point. I'll add that assertion.

Comment (lifflander) [reply src/vt/topos/location/location.impl.h:188]: Added

**Pull request merged**